TaskHouseApi/Service/AuthService.cs

```diff
@@ -0,0 +1,47 @@
+using TaskHouseApi.Model;
+using TaskHouseApi.Model.ServiceModel;
+using TaskHouseApi.Repositories;
+using System.Linq;
+using TaskHouseApi.DatabaseContext;
+
+namespace TaskHouseApi.Service
+{
+    public class AuthService : IAuthService
+    {
+        private IPasswordService passwordService;
+        private IUserRepository userRepository;
+
+        public AuthService(IPasswordService passwordService, IUserRepository userRepository)
+        {
+            this.passwordService = passwordService;
+            this.userRepository = userRepository;
+        }
+
+        public User Authenticate(LoginModel loginModel)
+        {
+            User potentialUser = (userRepository.RetrieveAll())
+                .SingleOrDefault(user => user.Username.Equals(loginModel.Username));
+
+            if (potentialUser == null || !isAuthenticated(loginModel, potentialUser))
+            {
+                return null;
+            }
+
+            return potentialUser;
+        }
+
+        private bool isAuthenticated(LoginModel loginModel, User potentialUser)
+        {
+            bool isPasswordGood = passwordService
```

Comment:
University-style-naming-discussion, but: isPasswordCorrect would be more appropriate in terms of describing the functionality (if I understand it right, that is)